python-backend/app/api/now_submissions/resources/application_resource.py

```diff
@@ -18,3 +27,42 @@ def get(self, application_guid):
             raise NotFound('Application not found')
 
         return application
+
+
+class ApplicationListResource(Resource, UserMixin, ErrorMixin):
+    @api.doc(
+        description='Get a list of applications. Order: receiveddate DESC',
+        params={
+            'page': f'The page number of paginated records to return. Default: {PAGE_DEFAULT}',
+            'per_page': f'The number of records to return per page. Default: {PER_PAGE_DEFAULT}',
+            'stuff': 'Comma-separated list of now submission applications to include in results.'
+        })
+    @requires_role_view_all
+    @api.marshal_with(PAGINATED_APPLICATION_LIST, code=200)
+    def get(self):
+        records, pagination_details = self._apply_filters_and_pagination(
+            page_number=request.args.get('page', PAGE_DEFAULT, type=int),
+            page_size=request.args.get('per_page', PER_PAGE_DEFAULT, type=int))
+
+        if not records:
+            raise BadRequest('Unable to fetch applications.')
+
+        data = records.all()
+
+
+        return {
+            'records': data,
+            'current_page': pagination_details.page_number,
+            'total_pages': pagination_details.num_pages,
+            'items_per_page': pagination_details.page_size,
+            'total': pagination_details.total_results,
+        }
+
+    def _apply_filters_and_pagination(self,
+                                        page_number=PAGE_DEFAULT,
+                                        page_size=PER_PAGE_DEFAULT):
+
+        filtered_query = db.session.query(Application) \
+            .order_by(desc(Application.receiveddate)) \
```

Comment:
Unnecessary escape here?

Also, I think we're almost always doing `ModelName.query` instead of `db.sessions.query(ModelName)`, so maybe change that for consistency?

Reply:
Yeah I noticed that! Refactored!